script.js

```diff
@@ -1,4 +1,346 @@
+  } else if (input == "lowest") {
+    isNormalMode = false;
+    mode = "gameMode";
+    return `You have selected Lowest mode. Player 1, it is your turn!`;
+  } else {
+    return `Please only enter normal or lowest!`;
+  }
+};
+
+// option for auto-combine mode
+var setAutoMode = function (input) {
+  if (input == "yes") {
+    autoMode = true;
+    mode = "knockoutModeSelect";
+    return `You have enabled Auto Mode. Do you wish to play knockout mode? Minumum players required: 3`;
+  }
+  if (input == "no") {
+    autoMode = false;
+    mode = "knockoutModeSelect";
+    return `You have disabled Auto Mode. Do you wish to play knockout mode? Minumum players required: 3`;
+  } else {
+    return `Please input either yes or no only!`;
+  }
+};
+
+// option for a knockout mode
+var setKnockoutMode = function (input) {
+  // minimum of 3 players needed
+  if (noOfPlayers <= 2) {
+    return `You need at least 3 players to play knockout mode!`;
+  } else if (input == "yes") {
+    knockoutMode = true;
+    mode = "gameModeSelect";
+    return `You have turned knockout mode on! Do you wish to play normal or lowest mode?`;
+  } else if (input == "no") {
+    knockoutMode = false;
+    mode = "gameModeSelect";
+    return `You have turned knockout mode off! Do you wish to play normal or lowest mode?`;
+  } else {
+    return `Please input either yes or no only!`;
+  }
+};
+
+var diceRoll = function () {
+  var randomDecimal = Math.random() * 6;
+  var randomDiceNumber = Math.floor(randomDecimal) + 1;
+  return randomDiceNumber;
+};
+
+var multipleDiceRoll = function (noOfPlayers, noOfDice, input) {
+  if (mode == "gameMode" || mode == "selectMode" || mode == "faceoff") {
+    while (currentPlayer < noOfPlayers) {
+      diceOutput = diceSelection(currentPlayer, noOfDice, input, noOfPlayers);
+      //currentPlayer only increases after player inputs their array
+      if (mode == "gameMode") {
+        currentPlayer = currentPlayer + 1;
+      }
+      return diceOutput;
+    }
+    mode = "faceoff";
+    faceoffOutput = faceOff(currentRoundCombinedNums);
+    mode = "numberOfDice";
+    //reset rolls array(for the round) at the end of game
+    currentRoundCombinedNums.length = 0;
+    return faceoffOutput;
+  }
+};
+
+var diceSelection = function (currentPlayer, noOfDice, input, noOfPlayers) {
+  //auto combine mode off, manual
+  if (autoMode == false) {
+    if (mode == "gameMode") {
+      console.log(currentPlayer);
+      index = 0;
+      while (index < noOfDice) {
+        diceRollsArray.push(diceRoll());
+        index = index + 1;
+      }
+      myOutputValue = `Player ${
+        currentPlayer + 1
+      },You have rolled ${diceRollsArray}! <br> 
+      Enter the order of the dice! The dice order starts from 0.`;
+      mode = "selectMode";
+      return myOutputValue;
+    } else if (mode == "selectMode") {
+      indexesArray = input.split("").map(Number);
+      combinedNum = 0;
```

Comment:
This is the first time that you have used combinedNum, so it would be clearer if you declared it as a new variable so that it is clearer. JS will create a new variable if it has not seen the variable name before but we would want to declare all our new variable for clarity